charmcraft/models/project.py

```diff
@@ -141,16 +141,17 @@ def _validate_name(cls, value: str) -> str:
             raise ValueError(
                 f"Library name invalid. Expected '[charm_name].[lib_name]', got {value!r}"
             )
-        if not re.fullmatch("[a-z0-9_]+", charm_name):
-            if "-" in charm_name:
-                raise ValueError(
-                    f"Invalid charm name in lib {value!r}. Try replacing hyphens ('-') with underscores ('_')."
-                )
+        # Accept python-importable charm names, but convert them to store-accepted names.
+        if "_" in charm_name:
+            charm_name = charm_name.replace("_", "-")
```

Comment:
WDYT about extending `tests/unit/models/test_project.py::test_create_valid_charm_lib` to exercise this conversion?  I don't see it tested elsewhere.

Reply:
Good point. Done!